src/s3/util/fileViewerManager.ts

```diff
@@ -2,30 +2,195 @@
  * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { OutputChannel } from 'vscode'
 import { ext } from '../../shared/extensionGlobals'
-import { S3FileNode } from '../explorer/s3FileNode'
-import { Window } from '../../shared/vscode/window'
+import { makeTemporaryToolkitFolder } from '../../shared/filesystemUtilities'
 import { showOutputMessage } from '../../shared/utilities/messages'
-import { OutputChannel } from 'vscode'
+import { Commands } from '../../shared/vscode/commands'
+import { downloadWithProgress } from '../commands/downloadFileAs'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { readablePath } from '../util'
+import { getStringHash } from '../../shared/utilities/textUtilities'
+
+const fs = require('fs')
+const SIZE_LIMIT = 4 * Math.pow(10, 6)
+
 export class S3FileViewerManager {
-    //private cache: Set<S3FileNode>
+    private cacheArn: Set<string>
```

Comment:
collections should be named with plural usually.
```suggestion
    private cacheArns: Set<string>
```